frontend/mariner2/handle_rpm.go

```diff
@@ -2,63 +2,46 @@ package mariner2
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	"path/filepath"
-	"runtime"
 	"sort"
-	"strconv"
 	"strings"
 
 	"github.com/Azure/dalec"
 	"github.com/Azure/dalec/frontend"
+	"github.com/Azure/dalec/frontend/rpm"
 	"github.com/moby/buildkit/client/llb"
 	"github.com/moby/buildkit/exporter/containerimage/image"
-	"github.com/moby/buildkit/frontend/dockerui"
 	gwclient "github.com/moby/buildkit/frontend/gateway/client"
-	"github.com/moby/buildkit/identity"
 )
 
 const (
-	marinerRef            = "mcr.microsoft.com/cbl-mariner/base/core:2.0"
-	toolchainImgRef       = "ghcr.io/azure/dalec/mariner2/toolchain:latest"
-	toolchainNamedContext = "mariner2-toolchain"
-
-	tookitRpmsCacheDir  = "/root/.cache/mariner2-toolkit-rpm-cache"
-	cachedRpmsName      = "mariner2-toolkit-cached-rpms"
-	marinerToolkitPath  = "/usr/local/toolkit"
-	marinerTdnfCacheDir = "/var/cache/tdnf"
+	marinerRef = "mcr.microsoft.com/cbl-mariner/base/core:2.0"
+
+	tdnfCacheDir  = "/var/cache/tdnf"
+	tdnfCacheName = "mariner2-tdnf-cache"
 )
 
-func defaultMarinerTdnfCahe() llb.RunOption {
-	return marinerTdnfCacheWithPrefix("")
+func defaultTndfCacheMount() llb.RunOption {
+	return tdnfCacheMountWithPrefix("")
 }
 
-// marinerTdnfCacheWithPrefix returns a run option that sets up a persistent cache for tdnf.
-// The tdnf cache is mounted at `[prefix]/var/cache/tdnf`.
-//
-// This makes it so that when tdnf needs to download packages, repodata, etc it will use the cache dir.
-// Repeated builds will benefit from this as the cache will be reused.
-func marinerTdnfCacheWithPrefix(prefix string) llb.RunOption {
-	return llb.AddMount(filepath.Join(prefix, marinerTdnfCacheDir), llb.Scratch(), llb.AsPersistentCacheDir("mariner2-tdnf-cache", llb.CacheMountShared))
+func tdnfCacheMountWithPrefix(prefix string) llb.RunOption {
+	return llb.AddMount(filepath.Join(prefix, tdnfCacheDir), llb.Scratch(), llb.AsPersistentCacheDir(tdnfCacheName, llb.CacheMountLocked))
```

Comment:
Could you explain why we are switching to `CacheMountLocked` here?

Reply:
tdnf is exiting due to locking issues with shared mode.

Reply:
To be clear, I'm not sure why it's not waiting for the lock, may be an issue with tdnf vs dnf.

Reply:
Ah got it. Maybe we could add a comment explaining that?

Reply:
Sure. Pushed that up, along with an extra commit to try and see if dnf is better here.
I'd like to use shared if we can.

Reply:
Yeah it just doesn't like being shared.
I've not seen these erorrs before now, but it seems like dnf appears to be somewhat racey in how it handles locks.

Reply:
I've removed that 2nd commit.